src/fhost_fpga.py

```diff
@@ -150,36 +157,63 @@ def __init__(self, host, katcp_port=7147, boffile=None, connect=True, config=Non
             self.min_load_time = None
             self.network_latency_adjust = None
 
+        self._pfb_of_counts = {}
+        self._ct_counts = {}
+        self._qdr_counts = {}
+
     @classmethod
     def from_config_source(cls, hostname, katcp_port, config_source):
         boffile = config_source['bitstream']
         return cls(hostname, katcp_port=katcp_port, boffile=boffile,
                    connect=True, config=config_source)
 
-    def ct_okay(self, sleeptime=1):
+    def ct_okay(self, wait_time=1):
         """
-        Is the corner turner working?
-        :return: True or False,
+        Checks if the corner turner working.
+        :param: wait_time : Float or None : If not None, fetch ct counter,
+            wait this long, and fetch a second value; Else, use last read value
+            from cache. Value is not cached if wait_time is None.
+        :return: bool : True, if corner tuner is okay
```

Comment:
There is no side-effect. Just always cache the last value read from the register, regardless of the "mode".